dvc/daemon.py

```diff
@@ -12,29 +12,34 @@
 
 logger = logging.getLogger(__name__)
 
-CREATE_NEW_PROCESS_GROUP = 0x00000200
-DETACHED_PROCESS = 0x00000008
+CREATE_NO_WINDOW = 0x08000000  # only available since Python 3.7 in subprocess.
 
 
 def _popen(cmd, **kwargs):
     prefix = [sys.executable]
     if not is_binary():
-        prefix += [sys.argv[0]]
-
+        main_entrypoint = os.path.join(
+            os.path.abspath(os.path.dirname(__file__)), "__main__.py"
+        )
+        prefix += [main_entrypoint]
```

Comment:
@iterative/dvc, do you know a better way to get the executable? In Windows, the `sys.argv[0]` is `.env/Scripts/dvc` which is just a link to `.env/Scripts/dvc.exe` build from the `console_scripts`.  So, we were calling it as `/path/to/python .env/Scripts/dvc` which is not a python file and the daemon was failing on Windows.  

In Linux/MacOS, based on how you invoke, it's either `__main__.py` or the `.env/bin/dvc`.

Reply:
@skshetry It is also important to not forget about our exe binary built by pyinstaller, IIRC `__main__.py` is not present there, but I might be mistaken.

Reply:
@efiop, It's inside the `if not is_binary():` block as before.

Reply:
@skshetry Ah, sorry, totally blinded that one. :facepalm: